catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/util/UiTestUtils.java

```diff
@@ -203,4 +206,28 @@ public static boolean comparePixelRgbaArrays(byte[] firstArray, byte[] secondArr
 		}
 		return true;
 	}
+
+	public static void waitForVariableValueEquals(UserVariable userVariable, double expectedValue, int timeout) {
```

Comment:
istead of the hard to read and comprehend version with the return; and assert in bottom, wouldnt it be nicer to have this function return a boolean, and rename it to something like "userVariableEqualsWithinTimeout" to then do something like:
assertTrue(userVariableEqualsValueWithinTimeout(uv, ev, t))